### ISSUE TITLE
SSH Remote cannot install on the host: Hanging when downloading with wget

### ISSUE THREAD

**Issue by alicanb** [microsoft/vscode-remote-release#4008]
Similar issue to https://github.com/microsoft/vscode-remote-release/issues/2582 

VSCode Version: 1.51.1
Commit: e5a624b788d92b8d34d1392e4c4d9789406efe8f
Date: 2020-11-11T01:11:34.018Z
Electron: 9.3.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0

- VSCode Version:
- Local OS Version: OSX 10.15.7
- Remote OS Version: Ubuntu 18.04.5
- Remote Extension/Connection Type: SSH 
- Remote SSH version v0.56

The log below:

```
[14:53:02.652] Log Level: 2
[14:53:02.653] remote-ssh@0.56.0
[14:53:02.653] darwin x64
[14:53:02.654] SSH Resolver called for "ssh-remote+hpc", attempt 1
[14:53:02.654] SSH Resolver called for host: hpc
[14:53:02.654] Setting up SSH remote "hpc"
[14:53:02.657] Acquiring local install lock: /var/folders/7f/p3g7fz4j0sd38gd0zdvhl1000000gr/T/vscode-remote-ssh-hpc-install.lock
[14:53:02.658] Looking for existing server data file at /Users/alican.bozkurt/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-hpc-e5a624b788d92b8d34d1392e4c4d9789406efe8f-0.56.0/data.json
[14:53:02.660] Using commit id "e5a624b788d92b8d34d1392e4c4d9789406efe8f" and quality "stable" for server
[14:53:02.661] Install and start server if needed
[14:53:02.662] Checking ssh with "ssh -V"
[14:53:02.671] > OpenSSH_8.1p1, LibreSSL 2.7.3

[14:53:02.674] askpass server listening on /var/folders/7f/p3g7fz4j0sd38gd0zdvhl1000000gr/T/vscode-ssh-askpass-e27f07703941ee7c9aa60a57d87461e45c307b92.sock
[14:53:02.674] Spawning local server with {"ipcHandlePath":"/var/folders/7f/p3g7fz4j0sd38gd0zdvhl1000000gr/T/vscode-ssh-askpass-f1912c7ade7ca28e99604ebfe94fc38c4cd9263a.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","52100","-o","ConnectTimeout=15","hpc"],"dataFilePath":"/Users/alican.bozkurt/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-hpc-e5a624b788d92b8d34d1392e4c4d9789406efe8f-0.56.0/data.json"}
[14:53:02.674] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/alican.bozkurt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.56.0/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)","VSCODE_SSH_ASKPASS_MAIN":"/Users/alican.bozkurt/.vscode/extensions/ms-vscode-remote.remote-ssh-0.56.0/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/7f/p3g7fz4j0sd38gd0zdvhl1000000gr/T/vscode-ssh-askpass-e27f07703941ee7c9aa60a57d87461e45c307b92.sock"}
--removing some stuff for privacy--
[14:53:04.247] > ready: 965dee299f4d
[14:53:04.324] > Linux 4.15.0-122-generic #124-Ubuntu SMP Thu Oct 15 13:03:05 UTC 2020
[14:53:04.324] Platform: linux
[14:53:04.403] > 965dee299f4d: running
[14:53:04.445] > Acquiring lock on /home/alican.bozkurt/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f/vscode-remote-lock.alican.bozkurt.e5a624b788d92b8d34d1392e4c4d9789406efe8f
[14:53:04.453] > Installing to /home/alican.bozkurt/.vscode-server/bin/e5a624b788d92b8d34d1392e4c4d9789406efe8f...
[14:53:04.453] > 965dee299f4d%%1%%
[14:53:04.459] > Downloading with wget
```

**Comment by Aldo-Aditiya** [comment]
Have you found a solution? I also encountered the same problem (it runs just fine last week in the same system that I'm trying to connect)

**Comment by roblourens** [comment]
I don't know what's wrong with the download, do you have a proxy or something else that might be interfering?

But I just added a setting that may help you work around this.

- Install the nightly version of the extension (uninstall the stable version first)
- Set `"remote.SSH.localServerDownload": "always"`, this will set it to download on the local side
- Try again

Let me know if it works

**Comment by jsalankey** [comment]
I am encountering the same issue running remote SSH to a Linux machine from a Windows machine. 

Local environment information:

Version: 1.51.1 (user setup)
Commit: e5a624b788d92b8d34d1392e4c4d9789406efe8f
Date: 2020-11-10T23:34:32.027Z
Electron: 9.3.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.18363

The remote OS I am connecting to is a RHEL 7 Linux machine. I tried fiddling with iptables and ports for the wget request to no avail. It hangs for a little while at the wget step, and ultimately fails to connect. I tried tweaking just about every SSH setting in the remote.


However, I did find a successful workaround: I rolled back the Remote SSH extension to 0.55.0 and it worked like a charm again.
